### DISCUSSION TITLE
git rebase interactive with windows editor tool fails - wrong path, home shortcut ~ is not working, path expansion wrong

### DISCUSSION THREAD

**Discussion by mruepp** [microsoft/WSL#6612] [General]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18363.1377]
Your Distribution version: Ubuntu Release: 20.04
Whether the issue is on WSL 2 and/or WSL 1: WSL2
```

# Steps to reproduce:

In .bashrc i create an alias:
```
alias subl='/mnt/c/Program\ Files/Sublime\ Text\ 3/subl.exe'
```

If i set  
```
[core]
        longpaths = true
        editor = /mnt/c/Program\\ Files/Sublime\\ Text\\ 3/subl.exe -n -w
```
in .gitconfig and then do a
```
git rebase -i origin/master
```
the todo list is empty because the path which was handed over to sublime is:
```
/home/<username>/path/to/repo/.git/rebase-todo
```
When I try to open gitconfig in the home folder by typing (remember alias in bashrc)
```
subl ~/.gitconfig
```
it opens an empty file because the path was not correctly expanded (I suspect the ~ also points to ```/home/<username>/```)

**WSL logs**: 

#  Expected behavior

The path conversion would work properly or the file system integration would be seamless

# Actual behavior

See in my description above

**Comment by therealkenc** [comment]
Variation #6266.

**Reply by mruepp** [reply]
This is not a variation of the mentioned Ticket. Its about how handover of path is happening in interaction of windows based software and linux based. So if there would be handover to a windows tool, the shell should either detect that the binary is windows, or, it should in any way use absolute path related to the host filesystem which is windows and convert on the fly between both formats. wslpath can not be used for seamless integration, who wants to write a script anytime you handover path. Also the tilde sign ~ is bash/unix native and this behaviour should be adressed accordingly in wsl: https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html
